lib/dry/struct/class_interface.rb

```diff
@@ -52,16 +52,15 @@ def check_schema_duplication(new_schema)
       private :check_schema_duplication
 
       def constructor_type(type = nil)
-        if type
-          @constructor_type = type
-        else
-          @constructor_type || :strict
-        end
+        return @constructor_type = type if type
+        @constructor_type
       end
 
       def schema
+        @schema = {} unless defined? @schema
```

Comment:
Nope. I'll try other possibilities but ```shema``` method is used in definition of method ```new``` (in ```default_attributes```). so it's a bit challenging. let's see what I can do.